File: src/test/java/io/bloco/faker/components/NameTest.java
package io.bloco.faker.components;

import org.junit.Before;
import org.junit.Test;

import io.bloco.faker.Faker;

import static org.hamcrest.Matchers.hasItem;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertThat;

public class NameTest {

    private Faker faker;

    @Before
    public void setUp() {
        faker = new Faker();
    }

    @Test
    public void firstName() {
        assertNotNull(faker.name.firstName());
    }

    @Test
    public void lastName() {
        assertNotNull(faker.name.lastName());
    }

    @Test
    public void prefix() {
        assertNotNull(faker.name.prefix());
    }

    @Test
    public void suffix() {
        assertNotNull(faker.name.suffix());
    }

    @Test
    public void title() {
        assertNotNull(faker.name.title());
    }

    @Test
    public void name() {
        assertNotNull(faker.name.name());
    }

    @Test
    public void nameWithMiddle() {
        assertNotNull(faker.name.nameWithMiddle());
    }

    @Test
    public void jobTitles() {
        assertThat(faker.name.jobTitles(), hasItem("Engineer"));
    }
}

File: src/test/kotlin/io/bloco/faker/components/NameTest.kt
package io.bloco.faker.components

import io.bloco.faker.Faker
import junit.framework.TestCase.assertFalse
import junit.framework.TestCase.assertNotNull
import junit.framework.TestCase.assertTrue
import org.junit.Test

class NameTest {
    private val faker = Faker()

    @Test
    fun firstName() {
        assertNotNull(faker.name.firstName())
    }

    @Test
    fun lastName() {
        assertNotNull(faker.name.lastName())
    }

    @Test
    fun prefix() {
        assertNotNull(faker.name.prefix())
    }

    @Test
    fun suffix() {
        assertNotNull(faker.name.suffix())
    }

    @Test
    fun title() {
        assertNotNull(faker.name.title())
    }

    @Test
    fun name() {
        assertNotNull(faker.name.name())
    }

    @Test
    fun name_ru() {
        val name = Faker("ru").name.name()
        assertNotNull(name)
        assertFalse(name.contains("#"))
    }

    @Test
    fun nameWithMiddle() {
        assertNotNull(faker.name.nameWithMiddle())
    }

    @Test
    fun jobTitles() {
        assertTrue(faker.name.jobTitles().contains("Engineer"))
    }
}
